### ISSUE TITLE
vhdl-ts-mode breaks indentation of conditional signal assignment and selected signal assignment statements

### ISSUE THREAD

**Issue by maehne** [gmlarumbe/vhdl-ts-mode#6]
Consider the following VHDL code formatted with `vhdl-mode`'s `vhdl-beautify-buffer`:

```vhdl
library ieee;
use ieee.std_logic_1164.all;

entity traffic_light_fsm is
  port (
    clock, reset, button : in  std_logic;
    leds                 : out std_logic_vector(2 downto 0));  -- red, yellow, green
end entity traffic_light_fsm;

architecture rtl of traffic_light_fsm is
  type state_type is (RED, YELLOW, GREEN, RED_YELLOW);
  signal state_reg, state_next : state_type;
begin  -- architecture rtl

  REG : process (clock, reset) is
  begin  -- process REG
    if reset = '1' then                 -- asynchronous reset (active low)
      state_reg <= RED;
    elsif rising_edge(clock) then       -- rising clock edge
      state_reg <= state_next;
    end if;
  end process REG;

  NSL : state_next <= RED_YELLOW when state_reg = RED and button = '1' else
                      GREEN  when state_reg = RED_YELLOW else
                      YELLOW when state_reg = GREEN and button = '1' else
                      RED    when state_reg = YELLOW else
                      state_reg;

  OL : with state_reg select
    leds <=
    "100" when RED,
    "110" when RED_YELLOW,
    "001" when GREEN,
    "010" when YELLOW,
    "---" when others;

end architecture rtl;
```

If I switch on `vhdl-ts-mode`, the VHDL code gets formatted like this when executing `vhdl-beautify-buffer`:

```vhdl
library ieee;
use ieee.std_logic_1164.all;

entity traffic_light_fsm is
  port (
    clock, reset, button : in std_logic;
    leds : out std_logic_vector(2 downto 0));  -- red, yellow, green
end entity traffic_light_fsm;

architecture rtl of traffic_light_fsm is
  type state_type is (RED, YELLOW, GREEN, RED_YELLOW);
  signal state_reg, state_next : state_type;
begin  -- architecture rtl

  REG : process (clock, reset) is
  begin  -- process REG
    if reset = '1' then                 -- asynchronous reset (active low)
      state_reg <= RED;
    elsif rising_edge(clock) then       -- rising clock edge
      state_reg <= state_next;
    end if;
  end process REG;

  NSL : state_next <= RED_YELLOW when state_reg = RED and button = '1' else
                                                                       GREEN when state_reg = RED_YELLOW else
                                                                                                         YELLOW when state_reg = GREEN and button = '1' else
                                                                                                                                                        RED when state_reg = YELLOW else
                                                                                                                                                                                    state_reg;

  OL : with state_reg select
  leds <=
  "100" when RED,
                "110" when RED_YELLOW,
                                     "001" when GREEN,
                                                     "010" when YELLOW,
                                                                      "---" when others;

end architecture rtl;
```

As you can see, each new line within conditional signal assignment and selected signal assignment statements gets increasingly indented making the code much less readable.

I am using GNU Emacs 29.3 with `vhdl-ts-mode` version 20240406.2155 (commit: eaeefce408085159cacf1ee9424fee20f20b6c39). The VHDL Mode version is 3.38.5 (2015-03-12).

**Comment by Rutherther** [comment]
Fixable by adding
```elisp
     ((and
       (node-is "waveforms")
       (or
        (parent-is "alternative_conditional_waveforms")
        (parent-is "alternative_selected_waveforms")))
      grand-parent 0) ; when else on next line or select multiple lines

     ((node-is "selected_waveforms") parent-bol vhdl-ts-indent-level)
```
to `vhdl-ts--indent-rules` properly - under vhdl.

The code you provided is then indented as
```vhdl
  NSL : state_next <= RED_YELLOW when state_reg = RED and button = '1' else
                      GREEN  when state_reg = RED_YELLOW else
                      YELLOW when state_reg = GREEN and button = '1' else
                      RED when state_reg = YELLOW else
                      state_reg;

  OL : with state_reg select
  leds <=
    "100" when RED,
    "110" when RED_YELLOW,
    "001" when GREEN,
    "010" when YELLOW,
    "---" when others;
```

**Comment by maehne** [comment]
@Rutherther: Thanks a lot for suggesting a fix for this issue! Did you notice that the `leds` assignment line isn't indented by a level with your fix?

Regarding the selected signal assignment statement, I personally would prefer if the assigned values would be all indented to be right of the assignment arrow similar to how it is done for the conditional signal assignments:

```vhdl
OL : with state_reg select
  leds <= "100" when RED,
          "110" when RED_YELLOW,
          "001" when GREEN,
          "010" when YELLOW,
          "---" when others;
```

Do you know by chance how to achieve this either in `vhdl-mode` or `vhdl-ts-mode`?

**Comment by Rutherther** [comment]
Yes, I noticed that. It was too much on me since it was late. I tried to fix it this morning and came up with a fix for your exact case, but noticed further discrepancies when with is on next line for example. And I am not sure how to make a good general fix for those, so I postponed it to think about it. To get `leds <=` shifted by indent, add this
```elisp
     ((and (node-is "simple_name")
           (parent-is "selected_concurrent_signal_assignment"))
      parent-bol vhdl-ts-indent-level)
```

I think this all achieves the behavior you want. I now get
```vhdl
OL : with state_reg select
  leds <= "100" when RED,
          "110" when RED_YELLOW,
          "001" when GREEN,
          "010" when YELLOW,
          "---" when others;
```
when I make sure to leave `"100" when RED` on the same line as `leds <=`

Also I noticed I forgot to copy part of the code to my previous message, I have fixed that. The code can also be seen in the PR #9 where I tried to fix this apart from other stuff that I didn't like about vhdl-ts-mode indentation.
